libr/util/str.c

```diff
@@ -499,13 +499,17 @@ R_API char *r_str_word_get0set(char *stra, int stralen, int idx, const char *new
 // XXX: Warning! this function is UNSAFE, check that the string has, at least,
 // idx+1 tokens.
 R_API const char *r_str_word_get0(const char *str, int idx) {
-	int i;
+	int i, ptrlen;
 	const char *ptr = str;
 	if (!ptr || idx < 0 /* prevent crashes with negative index */) {
 		return (char *)nullstr;
 	}
 	for (i = 0; i != idx; i++) {
-		ptr += strlen (ptr) + 1;
+		ptrlen = strlen (ptr);
+		ptr += ptrlen + 1;
+		if (!ptrlen) {
+			break;
```

Comment:
Shouldn't we `return NULL` here? If we reach the point where strlen(ptr) == 0, it means  we didn't find the ith entry of the string, so we should return NULL IMHO.